From 16e39c8d9f31a72f7c3baeaed695ff556d043edc Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 16 Aug 2018 11:28:26 +0800
Subject: [PATCH 4471/5242] MLK-19242 gpu: imx: dpu: framegen: Increase
 timeout when waiting for moving

commit  a6a920b3223883202405effd8537ac10cb2673d7 from
https://source.codeaurora.org/external/imx/linux-imx.git

It takes 33 milliseconds to scanout a frame when display refresh rate
is 30fps, and 42 milliseconds for 24fps.  So, if we wait for framegen
frame index moving, 30 milliseconds timeout value is not enough to
cover all reasonable display refresh rates.  50 milliseconds should not
be a bad choice.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
(cherry picked from commit 8564546882c940bb1852cbd132944b498f7ed102)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-framegen.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-framegen.c b/drivers/gpu/imx/dpu/dpu-framegen.c
index adc3e5f..981aefe 100644
--- a/drivers/gpu/imx/dpu/dpu-framegen.c
+++ b/drivers/gpu/imx/dpu/dpu-framegen.c
@@ -476,7 +476,7 @@ void framegen_read_timestamp(struct dpu_framegen *fg,
 void framegen_wait_for_frame_counter_moving(struct dpu_framegen *fg)
 {
 	u32 frame_index, line_index, last_frame_index;
-	unsigned long timeout = jiffies + msecs_to_jiffies(30);
+	unsigned long timeout = jiffies + msecs_to_jiffies(50);
 
 	framegen_read_timestamp(fg, &frame_index, &line_index);
 	do {
-- 
1.7.9.5

